sparse/client.go

```diff
@@ -74,65 +82,84 @@ func SyncFile(localPath string, remote string, timeout int, directIO bool) error
 	}
 	defer fileIo.Close()
 
-	client := newSyncClient(remote, timeout, localPath, fileSize, fileIo)
+	client := newSyncClient(remote, timeout, fileIo.Name(), fileSize, fileIo, directIO)
 
 	defer client.closeServer() // kill the server no matter success or not, best effort
 
-	err = client.syncFileContent(fileIo, fileSize, directIO)
+	syncStartTime := time.Now()
+	err = client.syncContent()
 	if err != nil {
 		log.Errorf("syncFileContent failed: %s", err)
 		return err
 	}
+	log.Debugf("finished sync for file: %v size: %v elapsed: %.2fs",
+		fileIo.Name(), fileSize, time.Now().Sub(syncStartTime).Seconds())
 
-	return err
+	return nil
 }
 
-func (client *syncClient) syncFileContent(file FileIoProcessor, fileSize int64, directIO bool) error {
-	if err := client.openServer(directIO); err != nil {
-		return fmt.Errorf("openServer failed, err: %v", err)
+func SyncSnapshot(snapshotName string, rw ReaderWriterAt, snapshotSize int64, remote string, timeout int) error {
```

Comment:
Rename this to `SyncContent(rw ReaderWriterAt, size int64, remote string, timeout int) error` so that we can remove the concept `snapshot` from this library. Besides, `SyncFile` can call the renamed function as well.

Reply:
Thank you. Implemented